### DISCUSSION TITLE
When to leave float mode? Which SOC source to use.

### DISCUSSION THREAD

**Discussion by ogurevich** [Louisvdw/dbus-serialbattery#727] [Ideas]
Classic constellation: several battery packs, probably a battery aggregator and, as a consequence of different values, a SmartShunt.
![sourceOfTruth](https://github.com/Louisvdw/dbus-serialbattery/assets/50322596/d217ef3f-53ef-414d-9b12-3c7ed23b973d)
There are currently two conditions in the driver when the float mode is leaves: the cell voltages vary greatly or the SOC of the battery falls below the set value.

What do you think of the consideration of whether the SOC value from the SmartShut would not be the better option at this point?

related code
```
 # allow max voltage again, if cells are unbalanced or SoC threshold is reached
                elif (
                    utils.SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT > self.soc
                    or voltageDiff >= utils.CELL_VOLTAGE_DIFF_TO_RESET_VOLTAGE_LIMIT
                ) and not self.allow_max_voltage:
                    self.allow_max_voltage = True
```

**Comment by mr-manuel** [comment]
I don't think that this should be part of this driver. If you are using multiple batteries, it's the battery aggregators job. If you are using one battery and a SmarShunt, then you can choose between them. If you want your own constellation, then you can aggregate the data yourself with the dbus-mqtt-battery.

**Reply by ogurevich** [reply]
on the one hand a BMS, that means its SOC is only 89% will never still in float mode although the battery is fully charged, on the other hand it is definitely not correct to compensate the error of the BMS's shunt by adjusting the driver.

the smart shunt and battery aggregator are passive elements. You provide the information to the VenusOS and should not take your own initiative and take control.

probably in this case it's ok the way it is.

**Reply by mr-manuel** [reply]
You can reduce the value in the config file for example to 80%. For the Daly BMS there is a new SoC reset function, for JKBMS this is planned, since it's a bit complicated. See https://github.com/Louisvdw/dbus-serialbattery/issues/703

**Comment by ogurevich** [comment]
... meanwhile here is an optional implementation of this idea. https://github.com/ogurevich/dbus-serialbattery/tree/ogdev
would you please take a look at it when you get a chance :)

**Comment by ogurevich** [comment]
just put it in #741
